### PR TITLE
refactor(*): unify & use noop

### PR DESCRIPTION
**Pull request by manudeli**

### CODE REVIEW

File: packages/query-core/src/utils.ts [hunk 1]
```diff
@@ -68,9 +68,7 @@ export type QueryTypeFilter = 'all' | 'active' | 'inactive'
 
 export const isServer = typeof window === 'undefined' || 'Deno' in globalThis
 
-export function noop(): undefined {
-  return undefined
-}
```
Comment (manudeli) [inline packages/query-core/src/utils.ts:73]: unified this noop return type from undefined to void

File: packages/query-core/src/queryObserver.ts [hunk 2]
```diff
@@ -349,7 +348,7 @@ export class QueryObserver<
     )
 
     if (!fetchOptions?.throwOnError) {
-      promise = promise.catch(noop)
+      promise = promise.catch(() => undefined)
```
Comment (TkDodo) [inline packages/query-core/src/queryObserver.ts:351]: why can’t we continue to use `noop` here ?

Comment (manudeli) [reply packages/query-core/src/queryObserver.ts:351]: If we use noop(function noop(): void {}, not function noop(): undefined {}), TypeScript error will be occured

Comment (TkDodo) [reply packages/query-core/src/queryObserver.ts:351]: at runtime, there’s no difference between a function that returns `undefined` and a function that returns `void`. would be good if we could keep `noop` here

Comment (manudeli) [reply packages/query-core/src/queryObserver.ts]: I agree, I reflect your review now in [92b4171](https://github.com/TanStack/query/pull/8294/commits/92b4171b1a7fdf5f4c179798201bb6b608bbd572)

Comment (TkDodo) [reply packages/query-core/src/queryObserver.ts]: nice 👍

**Review by TkDodo — APPROVED** [review]

File: packages/query-core/src/utils.ts (hunk 1 above)
Comment (manudeli) [reply packages/query-core/src/utils.ts:73]: Now @tanstack/query-core's noop's return type is undefined

File: packages/query-core/src/queryObserver.ts (hunk 2 above)
Comment (TkDodo) [reply packages/query-core/src/queryObserver.ts]: hmm TypeScript doesn’t like that :/

**Comment by manudeli** [comment]
I resolved it now😀

**Review by Scutua — APPROVED** [review]

**Review by Scutua — APPROVED** [review]

**Review by Scutua — APPROVED** [review]

**Pull request merged**